# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWindow.cpp
# Commit: c766791726f1
# Full Hash: c766791726f1bbce2bb7581c62ccc8b334e8c9e2
# Author: stransky <stransky@redhat.com>
# Date: 2025-02-06 04:08:03
# Description:
#   Bug 1943662 [Wayland] Don't call GdkCeiledScaleFactor() from compositor/render threads r=emilio
#   
#   - Mark SetEGLNativeWindowSize() calls Wayland only as we need to set explicitly wl_egl_window size on Wayland.
#   - Remove moz_container_wayland_egl_window_set_size() and use WaylandSurface directly.
#   - Remove plain call of FractionalScaleFactor() from nsWindow::SetEGLNativeWindowSize() as it used for logging only and move it to logging block.
# ==============================================================================

diff -r 12d7b2544b10 -r c766791726f1 widget/gtk/nsWindow.cpp
--- a/widget/gtk/nsWindow.cpp	Wed Feb 05 16:34:23 2025 +0000
+++ b/widget/gtk/nsWindow.cpp	Wed Feb 05 16:38:14 2025 +0000
@@ -9611,28 +9611,31 @@
 #ifdef MOZ_WAYLAND
 bool nsWindow::SetEGLNativeWindowSize(
     const LayoutDeviceIntSize& aEGLWindowSize) {
-  if (!GdkIsWaylandDisplay() || !mIsMapped) {
+  // SetEGLNativeWindowSize() is Wayland only call.
+  MOZ_ASSERT(GdkIsWaylandDisplay());
+
+  if (!mIsMapped) {
     return true;
   }
 
-  float scale = FractionalScaleFactor();
 #  ifdef MOZ_LOGGING
-  if (LOG_ENABLED()) {
+  if (LOG_ENABLED_VERBOSE()) {
+    float scale = FractionalScaleFactor();
     static uintptr_t lastSizeLog = 0;
     uintptr_t sizeLog =
         uintptr_t(this) + aEGLWindowSize.width + aEGLWindowSize.height + scale +
         aEGLWindowSize.width / scale + aEGLWindowSize.height / scale;
     if (lastSizeLog != sizeLog) {
       lastSizeLog = sizeLog;
-      LOG("nsWindow::SetEGLNativeWindowSize() %d x %d scale %f (unscaled "
+      LOGVERBOSE(
+          "nsWindow::SetEGLNativeWindowSize() %d x %d scale %f (unscaled "
           "%f x %f)",
           aEGLWindowSize.width, aEGLWindowSize.height, scale,
           aEGLWindowSize.width / scale, aEGLWindowSize.height / scale);
     }
   }
 #  endif
-  return moz_container_wayland_egl_window_set_size(
-      mContainer, aEGLWindowSize.ToUnknownSize());
+  return mSurface->SetEGLWindowSize(aEGLWindowSize.ToUnknownSize());
 }
 #endif
 